### PR TITLE
docs: Update README.md to have a consistent description of the project

### PR DESCRIPTION
**Pull request by markgrover**
### Summary of Changes
Updating README with a description to be consistent with other blog posts (like this one: https://eng.lyft.com/amundsen-lyfts-data-discovery-metadata-engine-62d27254fbb9)

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely, including a title prefix.
- [x] PR includes a summary of changes.
- [x] My commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"

### CODE REVIEW

**Comment by markgrover** [comment]
cc @feng-tao @Golodhros

File: README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@
   </a>
 </p>
 
-Amundsen is a metadata driven application for improving the productivity of data analysts, data scientists and engineers when interacting with data. It does that today by indexing data resources (tables, dashboards, streams, etc.) and powering a page-rank style search based on usage patterns (e.g. highly queried tables show up earlier than less queried tables). Think of it as Google search for data. The project is named after Norwegian explorer [Roald Amundsen](https://en.wikipedia.org/wiki/Roald_Amundsen), the first person to discover the South Pole.
+Amundsen is a data discovery and metadata engine for improving the productivity of data analysts, data scientists and engineers when interacting with data. It does that today by indexing data resources (tables, dashboards, streams, etc.) and powering a page-rank style search based on usage patterns (e.g. highly queried tables show up earlier than less queried tables). Think of it as Google search for data. The project is named after Norwegian explorer [Roald Amundsen](https://en.wikipedia.org/wiki/Roald_Amundsen), the first person to discover the South Pole.
```
Comment (feng-tao) [inline README.md:23]: `a Data Discovery And Metadata Platform` which matches https://eng.lyft.com/open-sourcing-amundsen-a-data-discovery-and-metadata-platform-2282bb436234

Comment (markgrover) [reply README.md:23]: haha, this one says `engine` https://eng.lyft.com/amundsen-lyfts-data-discovery-metadata-engine-62d27254fbb9

Comment (markgrover) [reply README.md:23]: 🤦

Comment (markgrover) [reply README.md:23]: @feng-tao, how strongly do you feel about engine vs. platform?

Comment (feng-tao) [reply README.md:23]: lgtm, I think engine is fine as well, sorry, in a meeting

Comment (markgrover) [reply README.md:23]: ok, then. Thanks in advance for merging!

**Pull request merged**